std/algorithm/sorting.d

```diff
@@ -151,12 +142,20 @@ bool isSorted(alias less = "a < b", Range)(Range r) if (isForwardRange!(Range))
         immutable limit = r.length - 1;
         foreach (i; 0 .. limit)
         {
-            if (!binaryFun!less(r[i + 1], r[i])) continue;
-            assert(
-                !binaryFun!less(r[i], r[i + 1]),
-                "Predicate for isSorted is not antisymmetric. Both" ~
-                        " pred(a, b) and pred(b, a) are true for certain values.");
-            return false;
+            static if(!isStrict)
+            {
+                if (!binaryFun!less(r[i + 1], r[i])) continue;
+                assert(!binaryFun!less(r[i], r[i + 1]),
+                        "Predicate for isSorted is not antisymmetric. Both" ~
```

Comment:
This is a good opportunity to make this message an `enum` as to avoid the `~` allocation at runtime.